danielMunteanu/src/main/java/danielCollectionTask/AbstractOrder.java

```diff
@@ -0,0 +1,86 @@
+package danielCollectionTask;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.UUID;
+
+public abstract class AbstractOrder implements Order {
+    private final String id;
+    private final LocalDateTime date;
+    private List<Item> items;
+
+    public AbstractOrder(List<Item> items) {
+        this.id = id();
```

Comment:
this.id = UUID.randomUUID().toString() how it looks?

Reply:
looks better, done